src/support/threads.cpp

```diff
@@ -139,15 +139,17 @@ void ThreadPool::initialize(size_t num) {
 }
 
 size_t ThreadPool::getNumCores() {
-#ifdef __EMSCRIPTEN__
+#if defined(__EMSCRIPTEN__) && !defined(__EMSCRIPTEN_PTHREADS__)
+  // In an Emscripten build without pthreads support, avoid the overhead of
+  // including support code for the below runtime checks.
```

Comment:
If you really to be sure to avoid the cost why not keep the `else/endif`?   That way you are not relying so much on linker magic to do what you hope it will do.

Reply:
Good point, thanks, I opened #6355